hknweb/candidate/views.py

```diff
@@ -467,6 +467,109 @@ def add_cands(request):
+        if not event_id:
+            messages.error(request, "Please input an event ID!")
+            return redirect(next_page)
+        event = Event.objects.filter(pk=event_id)
+        if not event:
+            messages.error(request, "Please input a valid event ID!")
+            return redirect(next_page)
+        event = event[0]
+    elif checkoff_type == "project":
+        project_name = request.POST.get("project_name", "")
+        project = CommitteeProject.objects.get(name=project_name)
+        projectDoneEntry = CommitteeProjectDoneEntry.objects.filter(committeeProject=project).first()
+        if projectDoneEntry is None:
+            messages.error(request, "Could not find a corresponding CommiteeProjectDoneEntry. Please make sure one is created for the project.")
+            return redirect(next_page)
+    elif checkoff_type == "dues":
+        dues_name = request.POST.get("dues_name", "")
+        due = DuePayment.objects.get(name=dues_name)
+        duesDoneEntry = DuePaymentPaidEntry.objects.filter(duePayment=due).first()
+        if duesDoneEntry is None:
+            messages.error(request, "Could not find a corresponding DuePaymentPaidEntry. Please make sure one is created for the due.")
+            return redirect(next_page)
+    elif checkoff_type == "forms":
+        forms_name = request.POST.get("forms_name", "")
+        form = CandidateForm.objects.get(name=forms_name)
+        formsDoneEntry = CandidateFormDoneEntry.objects.filter(form=form).first()
+        if formsDoneEntry is None:
+            messages.error(request, "Could not find a corresponding CandidateFormDoneEntry. Please make sure one is created for the form.")
+            return redirect(next_page)
+    
+
+    # Pre-screen and validate data
+    users = []
+    for row in mem_csv:
+        try:
+            memberdto = CandidateDTO(row)
+        except AssertionError as e:
+            messages.error(request, "Invalid CSV format. Check that your columns are correctly labeled and filled out for each row.")
+            return redirect(next_page)
+        user = User.objects.filter(first_name=memberdto.first_name, last_name=memberdto.last_name, email=memberdto.email)
+        if not user:
+            messages.error(request, "Could not find user " + memberdto.first_name + " " + memberdto.last_name + " with email " + memberdto.email + ". Please check these parameters again.")
+            return redirect(next_page)
+        users.append(user[0])
+
+    # Checkoff all
+    for user in users:
+        if checkoff_type == "event":
+            rsvp = Rsvp.objects.filter(event=event, user=user)
+            if rsvp.count() != 0:
+                rsvp = rsvp[0]
+                rsvp.confirmed = True
+            else:
+                rsvp = Rsvp.objects.create(user=user, event=event, confirmed=True)
+            rsvp.save()
+        elif checkoff_type == "project":
+            projectDoneEntry.users.add(user)
```

Comment:
For these 3 things, check if there is a duplicate
If there is, then don't add it (but you **don't** have to error out)

Reply:
Never mind, Django won't add a duplicate for you